### DISCUSSION TITLE
3-tier setup for Csla based project(s)

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1605] [General]
Hi All,
most of our applications are internal and we access our database(s) from web server without any 'application server'. We are looking into 3-tier deployments, is the 'Data Portal configuration' book of Using Csla series still relevant? 
Any help will be appreciated.
Regards

**Comment by rockfordlhotka** [comment]
It is relevant _in concept_. The specifics about how to set up and configure WCF are pretty irrelevant in today's world though.

Pretty much all the content in the book about how the data portal works, what it does for you, etc. is all the same.

Today though, there are some differences:

1. You'll most likely be relying on `MobileFormatter` instead of the older `BinaryFormatter` or `NetDataContractSerializer`
1. You'll most likely be using `HttpProxy` instead of `WcfProxy`

**Comment by Chicagoan2016** [comment]
@rockfordlhotka , thank you, I will share my progress here.
Kind Regards

**Reply by rockfordlhotka** [reply]
ProjectTracker uses the HttpProxy and has an app server project.
